msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMSALController.java

```diff
@@ -169,16 +177,51 @@ private Map getSilentParameters(MSALAcquireTokenSilentOperationParameters parame
         request.setForceRefresh(parameters.getForceRefresh());
         request.setLoginHint(parameters.getAccount().getUsername());
         request.setName(parameters.getAccount().getUsername());
-        request.setUserId(parameters.getAccount().getHomeAccountId());
+        request.setUserId(parameters.getAccount().getLocalAccountId());
```

Comment:
The goal here is to have the tenant-specific value, right? To support guest scenarios?

If **yes**, this is good.

If **no**, what's going on?

Reply:
We are mapping Account in Broker with upn currently. Let's discuss about this tommorow

Reply:
I wouldn't expect either of these values to be what you want then... consider adding a `username` property?

I'd suggest not calling this parameter `upn`, as that's a v1 claim which has been replaced by `preferred_username` in v2.

Per our schema doc

| Field name  |  Description |
|---|---|
| `home_account_id`  | Home account identifier for that authentication scheme. <br/><br/>For AAD/MSA:<br/>`<uid>.<utid>`<br/><br/>STS returns the clientInfo on both v1 and v2 for AAD. This value is combined from two fields in the client Info. It should be a unique account identifier across tenants (e.g. the value will be the same for user’s home tenant and a guest tenant). <br/><br/><br/>(There are other cases where this value might be different, but this affects `NTLM`/`Basic` scenarios and we can ignore it for our purposes)  |
| `local_account_id`  |  Original tenant specific account identifier. <br/><br/>Can be needed for legacy purposes. OID for AAD (in some unique cases subject instead of `OID`) and `CID` for MSA. Comes from the `id_token.oid` and falls back to `id_token.sub`.  |
| `username`  | The primary username that represents the user (corresponds to the preferred_username claim in in the `/v2.0` endpoint). It could be an email address, phone number, or a generic username without a specified format. Its value is mutable and might change over time. For MSA it's email. For `NTLM`, `NTLM` username.  |

Reply:
Thanks @iambmelt . This is should be `homeAccountId` in case of MSAL, changed it back. Currently in ad-accounts we are writing user id (oid) to AccountManager and using that to look up to get the account. So when making changes for joined account cache, if it's written with `homeAccountId`, we should be good.